internal/consensus/common_test.go

```diff
@@ -848,13 +850,24 @@ func randConsensusNetWithPeers(
 	}
 }
 
-func randGenesisState(
+func makeGenesisState(
```

Comment:
I guess this is mostly test plumbing anyway, but what do you think about pulling out an arguments type for this? E.g.,

```go
type genesisArgs struct {
   Validators int
   Power      int64
   Params     *types.ConsensusParams,
   Time       time.Time
}

st, vals := makeGenesisState(cfg, genesisArgs{
   Validators: 5,
   Power:      100,
   Params:     params,
})

if args.Time.IsZero() {
  args.Time = time.Now()
}
vs, privs := factory.ValidatorSet(args.Validators, args.Power)
doc := factory.GenesisDoc(cfg, args.Time, vs.Validators, args.Params)
// etc.
```

It doesn't change much, but makes the call sites less of a slurry. YMMV, though.